### PR TITLE
Warn on no valid corepops

### PR DESCRIPTION
**Pull request by willprice**

### CODE REVIEW

**Comment by willprice** [comment]
- Without `nullglob`, if no corepops are present in the directory, the script complains, setting this ensures the loop only runs over `supplied.corepop`.
- We output the error message to stderr only.
- I've converted the script to bash, inline with our recent changes.

File: find.sh [hunk 1]
```diff
@@ -12,4 +14,5 @@ for i in "supplied.corepop" `ls -1r ${OSNAME}/${ARCH}/*.corepop`; do
     fi 
 done
 # If no good executable can be found, exit with an error code.
+>&2 echo "No valid corepop found."
```
Comment (sfkleach) [inline find.sh]: Because the output is so verbose, I wonder if we need to make that message standout a bit more?

File: find.sh [hunk 2]
```diff
@@ -1,4 +1,6 @@
-#!/bin/sh
+#!/bin/bash
+set -euo pipefail
+shopt -s nullglob
```
Comment (sfkleach) [inline find.sh]: Love it.

Comment (willprice) [reply find.sh]: lol, I had to remove it... when I set this the shell word disappears and `ls -1r` is run, which just lists everything in the top level directory. I've removed `ls -1r`; is there any reason you were listing the files in reverse? If older corepops work, why not use them?

File: find.sh (hunk 1 above)
Comment (willprice) [reply find.sh]: Yeah?
It currently does this:

```
make build
make[1]: Entering directory '/home/will/projects/getpoplog/Seed'
mkdir -p "_build"
cp -rpP _download/Base/ _build/
make -C _build/Base build
make[2]: Entering directory '/home/will/projects/getpoplog/Seed/_build/Base'
mkdir -p _build
echo 'void test(){}' > _build/test.c
if `(cd _build; /usr/bin/gcc -no-pie -c test.c 2>&1)`; then make buildNoPie; else make buildWithoutNoPie; fi
make[3]: Entering directory '/home/will/projects/getpoplog/Seed/_build/Base'
test ! -z "-v -no-pie -Wl,-export-dynamic -Wl,-no-as-needed" # Protect against being called from the wrong context.
sed -e 's/{{{POP__CC_OPTIONS:[^}]*}}}/-v -no-pie -Wl,-export-dynamic -Wl,-no-as-needed/' < pop/src/syscomp/x86_64/asmout.p.template > pop/src/syscomp/x86_64/asmout.p
true
make[3]: Leaving directory '/home/will/projects/getpoplog/Seed/_build/Base'
make[2]: Leaving directory '/home/will/projects/getpoplog/Seed/_build/Base'
mkdir -p _build/poplog_base
( cd _build/Base; tar cf - pop ) | ( cd _build/poplog_base; tar xf - )
touch _build/Base.proxy # Create the proxy file to signal that we are done.
cp -rpP _download/Corepops _build/
/bin/rm -rf _build/Corepops/linux/x86_64/*
cp -p _build/poplog_base/pop/pop/corepop _build/Corepops/supplied.corepop
make -C _build/Corepops corepop
make[2]: Entering directory '/home/will/projects/getpoplog/Seed/_build/Corepops'
COREPOP=`./find.sh` && [ ! -z "$COREPOP" ] && cp -p "$COREPOP" corepop
No valid corepop found.
make[2]: *** [Makefile:20: corepop] Error 1
make[2]: Leaving directory '/home/will/projects/getpoplog/Seed/_build/Corepops'
make[1]: *** [Makefile:464: _build/Corepops.proxy] Error 2
make[1]: Leaving directory '/home/will/projects/getpoplog/Seed'
```

I suppose it's not that loud, I'll add something that makes it read like this:
```
##################################
# ERROR: No valid corepop found. #
##################################
```

File: find.sh (hunk 2 above)
Comment (sfkleach) [reply find.sh]: I was following a recommendation. But I have decided that the order needs to be curated, so I will be removing that very soon.

**Comment by sfkleach** [comment]
We have a failing CircleCi check - I suspect it is not significant - but just flagging it.

File: Makefile [hunk 3]
```diff
@@ -17,6 +17,5 @@ clean:
 # find the most recent working version and leave a copy in the file
 # called corepop at top-level.
 corepop:
-	COREPOP=`/bin/sh ./find.sh 2>/dev/null` && [ ! -z "$$COREPOP" ] && cp -p "$$COREPOP" corepop
+	COREPOP=`./find.sh` && [ ! -z "$$COREPOP" ] && cp -p "$$COREPOP" corepop
```
Comment (sfkleach) [inline Makefile:20]: Why allow the misleading error output to be visible?

Comment (willprice) [reply Makefile:20]: Because otherwise we flush away our error message that this commit adds.

**Comment by willprice** [comment]
> We have a failing CircleCi check - I suspect it is not significant - but just flagging it.

There is no CI config on this branch yet, that comes in the next PR which tests all corepops

File: Makefile (hunk 3 above)
Comment (willprice) [reply Makefile:20]: The `find.sh` script runs the corepops with 
```
`./$corepop ":sysexit()" 2>&1`
```
So the corepop output will be entirely captured in the command substitution.

**Pull request merged**